composer/algorithms/utils/augmentation_common.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from typing import Type, TypeVar, cast
+
+import torch
+from PIL.Image import Image as PillowImage
+from torchvision import transforms
+
+_InputImgT = TypeVar("_InputImgT", torch.Tensor, PillowImage)
+_OutputImgT = TypeVar("_OutputImgT", torch.Tensor, PillowImage)
+
+
+def image_as_type(image: _InputImgT, need_type: Type[_OutputImgT]) -> _OutputImgT:
+    """Creates a copy of an image-like object with a different type.
```

Comment:
Example calls? Discussion of expected dimensions?

Reply:
Clarification: this is a package private function that users are never supposed to call. I refrained from prefixing it with an underscore + added a passable docstring for consistency with out other package-private functions. But (evidently!) this is confusing, so I now prefix this function (and the one below) with leading underscores.

Also moved the file to `algorithms/utils` based on discussion with ravi.

I still can definitely make the docstring nicer if you want, but we should probably do this in lots of other places as well in this case.

Reply:
If an external user is developing their own methods, wouldn't they want to call this helper function?

Reply:
yes; I was thinking this is analogous to utility functions like sklearn `_check_X_y` that are often used when building your own estimators but not public, but now it looks like that function [is public](https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html?highlight=check_x_y). ...So I made it not prefixed with an underscore and gave it a nicer docstring.

Still no usage examples since it's small and straightforward (especially now), but I can add them if you want